common/leaderelection/leaderelection.go

```
@@ -40,7 +40,7 @@ type LeaderCallbacks struct {
 
 func NewElector(ctx context.Context, eventBusConfig eventbusv1alpha1.BusConfig, clusterName string, clusterSize int, namespace string, leasename string, hostname string) (Elector, error) {
 	switch {
-	case strings.ToLower(os.Getenv(common.EnvVarLeaderElection)) == "k8s":
+	case eventBusConfig.Kafka != nil || strings.ToLower(os.Getenv(common.EnvVarLeaderElection)) == "k8s":
```

In sensors/listener.go you mention kafka doesn't require leader election? If sources don't do leader election also, maybe return an error mentioning that in this function

---

Eventsources still need a leader election (under some circumstances, it's a little confusing). So unfortunately we still need this for the event source.

---

[Here's the logic](https://github.com/intuit-data-os/argo-events/blob/master/eventsources/eventing.go#L376-L393) that determines if the eventsource needs a leader election or not. The important thing to note is there are circumstances where an eventsource with a kafka eventbus requires a leader election.